packages/react-native-reanimated/Common/cpp/reanimated/CSS/interpolation/filters/FilterStyleInterpolator.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <reanimated/CSS/interpolation/PropertyInterpolator.h>
+#include <reanimated/CSS/interpolation/filters/FilterOperationInterpolator.h>
+#include <reanimated/CSS/utils/keyframes.h>
+
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace reanimated::css {
+
+struct FilterKeyframe {
+  const double fromOffset;
+  const double toOffset;
+  // If the value is nullopt, we would have to read it from the view style
+  // (in all other cases, both vectors will have the same number of elements of
+  // corresponding types - elements from the same index will form interpolation
+  // pairs)
+  const std::optional<FilterOperations> fromOperations;
+  const std::optional<FilterOperations> toOperations;
+  // Keyframe is discrete when there is a difference in order between from and to operations
```

Comment:
```suggestion
  // Keyframe is discrete when the fromOperations and toOperations aren't compatible
```

Maybe make more general? The current comment implies that the discrete interpolation happens only when the order is different but it is not the only case. The other case is the one in which filters are different, e.g.:

```
fromOperations: [{ blur: 10 }]
toOperations: [{ opacity: 0.5 }]
```